features/step_definitions/profiles.js

```diff
@@ -92,6 +95,26 @@ Then(/^I'm ?(not)* able to find the profile with id "([^"]*)"(?: with profile "(
             return callbackAsync(new Error(`Profile with id ${id} exists`));
           }
 
+          if (profile) {
+            const
+              compare = (a, b) => {
+                return a.roleId < b.roleId;
```

Comment:
Comparison functions must return consistent results, or else the sort result may be undefined.
Here, if `a == b`, then that function:
* returns `true` if called with `(a, b)`
* also returns `true` if called with `(b, a)`